modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/utils/settingsForm.es.js

```diff
@@ -75,13 +75,18 @@ export const getFilteredSettingsContext = ({
 					}
 
 					if (fieldName === 'dataSourceType') {
-						return {
+						const field = {
 							...updatedField,
 							name: generateName(name, updatedField),
 							predefinedValue: '["manual"]',
-							readOnly: true,
-							visible: false,
-						};
+						}
+
+						if (!name.includes('form_web')){
```

Comment:
Looks like the main goal of checking if the name doesn't includes the `form_web` substring can be the same as "this piece of code will not run on the Forms". However `name` can be changed. I don't have a suggestion for asserting if you are in the context of forms or not but We can try find.

Reply:
@matuzalemsteles, What do you think about it?

Reply:
Yes, I think the best thing here would be to add an API to the component where we can say what should be done. For example, let me simplify by showing a pseudo-code, there are two properties that kind of do this but you have to analyze which one to use to see what makes the most sense, `visibleProperties` and `unsupportedProperties`.

```
unsupportedProperties={['dataSourceType']}
```

If there is a type `dataSourceType` in `unsupportedProperties` we can ignore this conditional, we do not always need to define `name` and `predefinedValue` because this already comes from the backend, but if there is `dataSourceType` this can be done (Maybe because of DE but not for Forms).

Reply:
We are going to postpone this change to use an API after we remove the Data Engine's feature flag, once this specific area is common to the old and the new js code and an API would force us to change both.